src/core/dbus-service.c

```diff
@@ -91,6 +94,68 @@ static int property_get_exit_status_set(
         return sd_bus_message_close_container(reply);
 }
 
+int bus_service_method_bind_mount(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        _cleanup_free_ char *error_path = NULL;
+        int read_only, make_file_or_directory;
+        const char *dest, *src;
+        char **inaccessible_paths = NULL;
+        Unit *u = userdata;
+        ExecContext *c;
+        int r;
+
+        assert(message);
+        assert(u);
+
+        r = mac_selinux_unit_access_check(u, message, "start", error);
+        if (r < 0)
+                return r;
+
+        r = sd_bus_message_read(message, "ssbb", &src, &dest, &read_only, &make_file_or_directory);
+        if (r < 0)
+                return r;
+
+        if (!path_is_absolute(src) || !path_is_normalized(src))
+                return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Source path must be absolute and not contain ../.");
```

Comment:
"Source path must be absolute and normalized (no . or .. components)" ?

Reply:
Fixed both - they were copied from bus_machine_method_bind_mount, fixing those too